### PR TITLE
split crate into "dlpi" and "libdlpi-sys"

### PR DESCRIPTION
**Pull request by jclulow**
Split out the raw library interface into a libdlpi-sys crate that can be used by other wrappers, and in the process adjust a few of the function and constant names to match the C source and manual pages.

Also fix run_tests.sh so that it builds the tests as the current user before escalating to root to run them.

### CODE REVIEW

**Comment by jclulow** [comment]
I was able to consolidate some of my DLPI code into using the new `libdlpi-sys` crate here: https://github.com/oxidecomputer/etherboot-server/commit/a4b3d2ac51fef9afad2996e89ed9bce773b34b3b

**Comment by jclulow** [comment]
I suspect the toolchain in use is too old for current crates, and there is no lock file.  I don't think there's anything about this that needs nightly -- should I switch it to latest stable?

**Comment by rcgoodfellow** [comment]
Sorry I did not notice this until now! Taking a look.

**Comment by rcgoodfellow** [comment]
> I suspect the toolchain in use is too old for current crates, and there is no lock file. I don't think there's anything about this that needs nightly -- should I switch it to latest stable?

Yeah, that would be great. I don't think we need nightly anymore, and certainly not this very old nightly.

File: dlpi/src/sys.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-use num_enum::TryFromPrimitive;
```
Comment (rcgoodfellow) [inline dlpi/src/sys.rs:1]: Should this file just go away in favor of libdlpi-sys/src/lib?

Comment (jclulow) [reply dlpi/src/sys.rs:1]: Almost certainly!  I think I forgot to delete it basically.  I'll give that a swing.

**Comment by jclulow** [comment]
@rcgoodfellow I think this is good to (squash and) merge now?

**Review by rcgoodfellow — APPROVED** [review]
Yup, LGTM. Thanks!

**Pull request merged**